### PR TITLE
Support for setting ignore as string

### PR DESCRIPTION
**Pull request by kinetifex**
In order for configurations to be set up in JSON, this adds the ability to set the ignore option as a string which will be used as the argument for a RegExp instance.

### CODE REVIEW

File: packages/babel-plugin-short-css-vars/test/fixtures/short-css-vars/ignore-string/options.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "plugins": [
+    ["../../../../", {
+      "ignore": "two"
```
Comment (msluther) [inline packages/babel-plugin-short-css-vars/test/fixtures/short-css-vars/ignore-string/options.json:4]: maybe put something in the string to make it obvious that it's converted to regex?  maybe something like 
```suggestion
      "ignore": "^.+-two"
```

File: packages/postcss-short-css-vars/index.test.js [hunk 2]
```diff
@@ -68,12 +68,21 @@ describe('postcss-short-css-vars', () => {
   });
 
   describe('ignore', () => {
+    it('ignores variable names via string', () => {
+      const input = ':root{--custom-var-one: red; --custom-var-two: 1.2rem}' +
+        '.my-class{ color: var(--custom-var-one, var(--custom-var-two))}';
+      const expected = ':root{--custom-var-one: red; --1tn4ykf: 1.2rem}' +
+        '.my-class{ color: var(--custom-var-one, var(--1tn4ykf))}';
+      const result = run(input, { ignore: 'one$' });
```
Comment (msluther) [inline packages/postcss-short-css-vars/index.test.js:76]: since you have `$` in there, maybe show a input->output where `one` exists but not at the end of the string

Comment (msluther) [reply packages/postcss-short-css-vars/index.test.js:76]: same for the rule below.

File: packages/short-css-vars/index.js [hunk 3]
```diff
@@ -22,20 +22,28 @@ const defaultFormatter = name => stringHash(name).toString(36);
...
+
+  //  Normalize ignore string/regexp to a function
+  if (ignoreFn && !(ignoreFn instanceof Function)) {
+    let ignore = options.ignore;
+    if (typeof ignore === 'string') {
+      ignore = new RegExp(ignore);
+    }
+    if (ignore instanceof RegExp) {
+      ignoreFn = str => ignore.test(str);
+    }
```
Comment (msluther) [inline packages/short-css-vars/index.js:39]: should you error out at this point with a good error message? It's not a function or a string or a regexp.

File: packages/short-css-vars/index.test.js [hunk 4]
```diff
@@ -72,12 +72,21 @@ describe('ShortCssVars', () => {
...
+        assume(result).equals(expected);
+      });
+
       it('ignores variable names via RegExp', () => {
         const input = ':root{--custom-var-one: red; --custom-var-two: 1.2rem}' +
           '.my-class{ color: var(--custom-var-one, var(--custom-var-two))}';
         const expected = ':root{--custom-var-one: red; --1tn4ykf: 1.2rem}' +
           '.my-class{ color: var(--custom-var-one, var(--1tn4ykf))}';
-        const result = run(input, { ignore: /one/ });
+        const result = run(input, { ignore: new RegExp('one$') });
```
Comment (msluther) [inline packages/short-css-vars/index.test.js:89]: why this change? does `/one/` not work for some reason?

File: packages/postcss-short-css-vars/index.test.js (hunk 2 above)
Comment (kinetifex) [reply packages/postcss-short-css-vars/index.test.js:76]: I only added the $ to demonstrate it does in fact get used as regexp. I'll adjust this to a similar pattern suggested in your [other comment](https://github.com/godaddy/short-css-vars/pull/1#discussion_r481321093).

File: packages/short-css-vars/index.test.js (hunk 4 above)
Comment (kinetifex) [reply packages/short-css-vars/index.test.js:89]: I think I was just trying some things, I'll change it back to slashes

File: packages/short-css-vars/index.js [hunk 5]
```diff
@@ -13,6 +13,30 @@ const reRef = /(,?\s*var\(\s*--)([\w-]+)(,\s*var.+)?/g;
  */
 const defaultFormatter = name => stringHash(name).toString(36);
 
+/**
+ * Normalize ignore string/regexp to a function
+ *
+ * @param {RegExp|string|function} ignore - Rule to ignore certain variable names
+ * @returns {(function(*): boolean)} fn
```
Comment (msluther) [inline packages/short-css-vars/index.js:20]: should this be?

```suggestion
 * @returns {(function(string): boolean)} fn
```
